### DISCUSSION TITLE
Question: Moving widgets

### DISCUSSION THREAD

**Discussion by cpiber** [Textualize/textual#3748] [Q&A] [answered]
Is there an API to move widgets beyond `Widget.move_child`? Or a way to "share" it (show in several places)?

I currently have this situation: There's an application-wide log that I want to display using `RichLog`. I can easily show the log on one screen. But I want to give the user to change to a sub-screen with a different layout.

I tried having it as a global and yielding it from multiple places, but that breaks layout and styling. If I instead remove it from it's old place in `on_mount` and `mount` it in its new place, styling and layout works as expected, but the widget is broken (specifically, it does not properly update anymore, and scrollbars do not appear).

I'm probably overlooking some ergonomic way to solve this instead of moving widgets in the DOM, do you have any suggestions?

**Comment by davep** [comment, marked as answer]
No, as you've found, the only method for moving a widget is changing its position amongst its siblings.

If you're wanting to keep a single log while having other parts of the display change, I'd probably approach it one of a couple of days:

- Just use a single screen, split between the log and all the other things you want to show, and encapsulate the other things you want to show in their own parent, making a compound widget, and them `remove` and `mount` that as required (a variation on this would be to use `ContentSwitcher` to manage the other widgets).
- Keep the approach you have but perhaps retain a central structure of the data you want to display in the log, and populate the log widget when the new screen is shown.

**Reply by cpiber** [reply]
A split would require the log to always be the outer-most element, which does not work well for my intentions.

Then I guess I'll have to track the lines myself, and add the lines `on_mount` everytime. Thanks for your input.
